### ISSUE TITLE
Incorrect Detokenisation for Special Tokens by SentencePiece-based Tokenizers

### ISSUE THREAD

**Issue by abheesht17** [keras-team/keras-nlp#797]
Ran the `detokenize` method for all the tokenisers which KerasNLP currently offers. Check out `DebertaV3Tokenizer`, `FNetTokenizer` and `AlbertTokenizer` outputs here: https://colab.research.google.com/drive/11kpyA_CwmZ0mj1fpmSU4BooovIsRKuYn?usp=sharing.

All three are SentencePiece-based tokenisers. Special tokens are detokenised to an empty string.

**Comment by ADITYADAS1999** [comment]
This is for new contributors ?

**Comment by ADITYADAS1999** [comment]
Can I work with this issue??

**Comment by abheesht17** [comment]
@ADITYADAS1999 - I am not sure how to proceed here. Will have to think of how to resolve this issue!

**Comment by ADITYADAS1999** [comment]
> @ADITYADAS1999 - I am not sure how to proceed here. Will have to think of how to resolve this issue!

I think 🤔 it little bit tricky.

**Comment by mattdangerw** [comment]
I believe that sentence piece has the notion of "control characters" which are added after tokenization and never intended to be decoded, and "user symbols" which can appear in the source text and should probably decode.

https://github.com/google/sentencepiece/blob/master/doc/special_symbols.md

If the issue we are seeing is "control characters" being detokenized to an empty string, this may be "works as intended" for the sentencepiece library.

But it definitely is a little awkward for us (inconsistent detokenize behavior). If there was an easy way to fix this up could be nice, but I'm not sure there will be.

**Comment by abheesht17** [comment]
Yeah, exactly. The SentencePiece folks really need to make it easier to modify the proto :(
